chrome/content/coniks/coniks.js

```diff
@@ -0,0 +1,118 @@
+this.EXPORTED_SYMBOLS = ["coniks"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("chrome://otr/content/coniks/context.js");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/otr.js");
+
+function accountname(acc) {
+  let account, protocol;
+  if (acc.normalizedName !== undefined &&
+    acc.protocol.normalizedName !== undefined) {
+    account = acc.normalizedName;
+    protocol = acc.protocol.normalizedName;
+  } else {
+    account = acc.account;
+    protocol = acc.protocol;
+  }
+  return `${account}@${protocol}`;
+}
+
+var coniks = {
+  isEnable: false,
+
+  init: function() {
+    coniks.setPrefs();
+    coniks.isEnabled = coniks.prefs.getBoolPref("enabled");
+    if (!coniks.isEnabled)
+      return;
+
+    this._contexts = new Map();
+    for (let acc of helpers.getAccounts()) {
+      coniks._contexts.set(accountname(acc), new AccountContext({
+        account: acc.normalizedName,
+        protocol: acc.protocol.normalizedName
+      }));
+    }
+    return coniks.loadFiles();
+  },
+
+  destroy: function() {
+    if (!coniks.isEnabled)
+      return;
+  },
+
+  onAccountCreated: function(acc) {
+    let account = acc.normalizedName;
+    let protocol = acc.protocol.normalizedName;
+    let fingerprint = otr.privateKeyFingerprint(account, protocol);
+    coniks._contexts.set(accountname(acc), new AccountContext({
+      account: account,
+      protocol: protocol,
+      fingerprint: fingerprint
+    }));
+    return coniks.store(coniks._contexts);
+  },
+
+  prefs: null,
+
+  setPrefs: function() {
+    let branch = "extensions.otr.coniks.";
+    let coniksPrefs = {
+      enabled: false,
+      serverAddress: "111.221.102.190",
+      serverPort: 3000
+    };
+    let defaults = Services.prefs.getDefaultBranch(branch);
+    Object.keys(coniksPrefs).forEach(function(key) {
+      if (typeof coniksPrefs[key] === "boolean") {
+        defaults.setBoolPref(key, coniksPrefs[key]);
+      } else {
+        defaults.setCharPref(key, coniksPrefs[key]);
+      }
+    });
+    coniks.prefs = Services.prefs.getBranch(branch);
+  },
+
+  loadFiles: function() {
```

Comment:
Or, wait, why have this? `coniks.load()` can just use `coniks._contexts` directly.